### PR TITLE
GFF Amharic Keyboard v1.6

### PR DESCRIPTION
**Pull request by dyacob**
Address some special cases for phoneme boundaries. Avoids producing a .js file for touch layout.

### CODE REVIEW

**Comment by keyman-server** [comment]
Keyboards: This pull request is not from a keyboards maintainer and will not automatically be built. The build must still be passed before it can be merged. Ask one of the team members to make a manual build of this PR.

**Comment by LornaSIL** [comment]
@dyacob Can you please update the version number in these file?

* HISTORY.md (add to history)
* README.md (change version number)
* gff_amharic.kpj (remove the version number so it doesn't need updating next time)

Also, since you updated the welcome.htm file does the help file need updating?

**Comment by keyman-server** [comment]
Keyboards: This pull request is not from a keyboards maintainer and will not automatically be built. The build must still be passed before it can be merged. Ask one of the team members to make a manual build of this PR.

**Comment by LornaSIL** [comment]
@dyacob the .kps has a line:
```    <CompilerWarningsAsErrors>False</CompilerWarningsAsErrors>
```
Can you change that value to ```True```? As far as I can tell, there aren't warnings any more and it would be cleaner to make that True.

File: release/gff/gff_amharic/source/gff_amharic.kps [hunk 1]
```diff
@@ -189,7 +189,7 @@
       <FileType>.ttf</FileType>
     </File>
     <File>
-      <Name>..\build\gff_amharic.js</Name>
+      <Name>gff_amharic.js</Name>
```
[the remark this answers is not in the document]
Comment (dyacob) [reply release/gff/gff_amharic/source/gff_amharic.kps]: Thanks @darcywong00 , I tried adding it back but then I get an error dialog that the file cannot be found and Keyman Developer will not save the edit to the .kps file.  This is probably why I changed the path.  If I create the folder "build" and move the generated gff_amharic.js there, then Developer will save the .kps and build, but then it is back to error message about version conflicts.

**Comment by dyacob** [comment]
@LornaSIL I don't see this tag in the .kps file, I wonder how you are seeing it but now I.  I tried adding it under the &lt;Options&gt; but Keyman Developer removes it.  I poked around a few other .kps files in github but can't find an example of its use. thanks.

**Comment by darcywong00** [comment]
It's at the top of the .kpj project file
```
  <Options>
    <BuildPath>$PROJECTPATH\build</BuildPath>
    <CompilerWarningsAsErrors>False</CompilerWarningsAsErrors>
    <WarnDeprecatedCode>True</WarnDeprecatedCode>
  </Options>
```

File: release/gff/gff_amharic/source/gff_amharic.kps (hunk 1 above)
[the remark this answers is not in the document]
Comment (darcywong00) [reply release/gff/gff_amharic/source/gff_amharic.kps]: If you manually edit the .kpj file as below, maybe modify the .kmn section too
```
    <File> 
      <ID>id_3793695f951846043f549dfecde24dab</ID> 
      <Filename>gff_amharic.kmn</Filename> 
      <Filepath>source\gff_amharic.kmn</Filepath> 
      <FileVersion>1.6</FileVersion> 
      <FileType>.kmn</FileType> 
```
to add a FileVersion

**Comment by keyman-server** [comment]
Keyboards: This pull request is not from a keyboards maintainer and will not automatically be built. The build must still be passed before it can be merged. Ask one of the team members to make a manual build of this PR.

**Comment by dyacob** [comment]
Thanks @LornaSIL &amp; @darcywong00 I have it building now without issue.  I've installed and tested the resulting .kmp file. Earlier I had been trying to build the .kps directly in Developer which I think lead to the issues I had, when I should have been working from the .kpj.

**Comment by LornaSIL** [comment]
@dyacob 

As @darcywong00 said, the setting I wanted changed is in the .kpj file, not the .kps file. Please change: False to True.
```
<Options>
    <BuildPath>$PROJECTPATH\build</BuildPath>
    <CompilerWarningsAsErrors>False</CompilerWarningsAsErrors>
    <WarnDeprecatedCode>True</WarnDeprecatedCode>
  </Options>
```
You can either manually edit the .kpj file or you can change it in KM developer by going to **Project / Project Settings** and select **Treat compiler hints and warnings as errors**

One other thing that I should have caught before. All "Tavultesoft" references, copyright, etc should be changed to "SIL International." (I see it in the .kmn, kps, kpj, README.md, and the help file.

The help file has a lot of changes needed:
 * Change contact from ```<a target="_blank" href="http://www.tavultesoft.com/contact.php">``` to ```https://keyman.com/contact/```
 * ```Tavultesoft graciously``` to ```SIL International graciously```
 * Copyright at bottom ```Copyright 1997-2018 Ge'ez Frontier Foundation and Tavultesoft.``` change Tavultesoft to SIL International
* Delete the last paragraph that starts 'THIS SOFTWARE IS PROVIDED BY TAVULTESOFT PTY LTD "AS IS"' as that is the old license

**Comment by dyacob** [comment]
Thanks for the review, I believe I have all of the identified changes.

**Comment by keyman-server** [comment]
Keyboards: This pull request is not from a keyboards maintainer and will not automatically be built. The build must still be passed before it can be merged. Ask one of the team members to make a manual build of this PR.

**Pull request merged**